docs/02-app/02-api-reference/01-components/form.mdx

```diff
@@ -0,0 +1,311 @@
+---
+title: <Form>
+description: Learn how to use the `<Form>` component to handle form submissions and search params updates with client-side navigation.
+---
+
+The `<Form>` component extends the HTML `<form>` element to provide **prefetching**, **client-side navigation**, and **progressive enhancement**. It's useful for forms that update URL search params on submission as it reduces boilerplate code needed while improving the navigation experience.
```

Comment:
We need to be clearer on what we mean by "prefetching" - if at all. See https://github.com/vercel/next.js/pull/68102#discussion_r1696987549 and [Seb's message](https://vercel.slack.com/archives/C03KAR5DCKC/p1722395172200909?thread_ts=1722385899.178839&cid=C03KAR5DCKC).

Reply:
My understanding is that we prefetch the route defined in the action (which might be wrong, still learning this one). What Zack was referencing there was basically that we _also_ need that PR to merge to have this story be true. I was able to confirm that case because I wasn't yet about to prefetch `/search` when trying to refactor Next.js Commerce to use `<Form>`.

Reply:
I wonder if we drop progressive enhancement from the intro – not because it's not helpful, but isn't it really around server actions more than anything? My understanding if you don't _lose_ this progressive enhancement when you sprinkle `<Form>` on top, but is there anything additional it's doing for you?

Reply:
Yeah, waiting for this one: https://github.com/vercel/next.js/pull/68340

My understanding it prefetches shared UI (`loading.js` and `layout.js`) and not the page itself. I don't think it could prefetch the page as it depends on search params which are only known on submission. 

re. progressive enhancement, the reason I included it is because of FUD. For example, I listened to a podcast from a popular Next.js alternative who claimed that `next/link` didn't work without JS loading. We're doing prefetching and client-side navigation while maintaining progressive enhancement.